drivers/usb/usbport/endpoint.c

```diff
@@ -1260,8 +1263,11 @@ USBPORT_ReopenPipe(IN PDEVICE_OBJECT FdoDevice,
 
             KeReleaseSpinLockFromDpcLevel(&FdoExtension->MiniportSpinLock);
         }
+        else
+        {
+            KeReleaseSpinLockFromDpcLevel(&Endpoint->StateChangeSpinLock);
```

Comment:
Why not pull it before the if?

Reply:
Because `Endpoint->StateLast` is protected by StateChangeSpinLock.